superset/assets/javascripts/modules/visUtils.js

```diff
@@ -1,9 +1,13 @@
 export function getTextWidth(text, fontDetails = '12px Roboto') {
   const canvas = document.createElement('canvas');
   const context = canvas.getContext('2d');
-  context.font = fontDetails;
-  const metrics = context.measureText(text);
-  return metrics.width;
+  if (context) {
+    // Won't work outside of a browser context (ie unit tests)
+    context.font = fontDetails;
```

Comment:
nit: `context = {...context, font: fontDetails}`

Reply:
I tested it in my console:
```
> test_obj = { my_func: () => 'test'}
{ my_func: [Function: my_func] }

> new_test_obj = {...test_obj}
{ my_func: [Function: my_func] }

> new_test_obj
{ my_func: [Function: my_func] }

> new_test_obj.my_func()
'test'

```
It seems like it works, but no biggie if you want to keep it as is. I'm not 💯% sure it won't break at runtime.

Reply:
I meant prototype funcs